lib/matplotlib/text.py

```diff
@@ -1177,6 +1178,10 @@ def set_rotation(self, s):
             The rotation angle in degrees in mathematically positive direction
             (counterclockwise). 'horizontal' equals 0, 'vertical' equals 90.
         """
+        if isinstance(s, str):
```

Comment:
The two separate checks lead to slightly confusing messages as they don't mention the respective other valid parameters, e.g. 

> ValueError: '30' is not a valid value for rotation; supported values are 'vertical', 'horizontal'

does not mention numbers.

I'd go for a manual check and message here:

```
if s is not None and not isinstance(s, numbers.Real) and s not in ['vertical', 'horizontal']:
    Raise ValueError("rotation must be 'vertical', 'horizontal' or a number, not {s!r}"]
```